### ISSUE TITLE
FNX2-15152 ⁃ [Bug] Firefox Beta and Nightly fail to download anything from any website

### ISSUE THREAD

**Issue by parvinderandroid** [mozilla-mobile/fenix#12297]
## STEPS TO REPRODUCE
1. Open Firefox (Beta or Nightly)
1. Open https://github.com/TeamNewPipe/NewPipe/releases/tag/v0.19.5
1. Click on Assets
1. Click on NewPipe_v0.19.5.apk
1. Start Download

### EXPECTED BEHAVIOUR
1. Downloading begins
1. Downloading finishes

### ACTUAL BEHAVIOUR
1. Downloading immediately fails

### DEVICE INFORMATION
* Android device: Huawei Honor 8 Pro (DUK-L09) - Android 9.0
* Fenix version: 79.0.0-beta.4 & Nightly 200705 06:01

### VIDEO OF ISSUE:
https://i.imgur.com/H9Qwxfc.mp4

**Comment by Augures01** [comment]
Can't reproduce with latest Nightly 200705 06:01 / Samsung S6 Edge, Android 7.0

**Comment by RealGJ** [comment]
Unable to reproduce with : Nightly 200705 06:01 (Build #2015750195)
AC: 49.0.20200702190156, bafa9299b
GV: 80.0a1-20200702094606
AS: 61.0.7

Device : Xiaomi Redmi Note 4 Android 7.0

**Comment by MinionAttack** [comment]
In my case I cannot download anything from any site, everything fails. This has been happening to me for a long time, even before moving Firefox Preview to Firefox Nightly.

I have tried removing all the plugins, checking and authorizing all the permissions, nothing works.

If I want to download something I have to use Firefox Focus. It's frustrating.

Xperia Z3 on Android 6.0.1

**Comment by cadeyrn** [comment]
@MinionAttack In that case you should open your own issue because not being able to download anything from any website is a different problem than not being able to download from one specific website.

**Comment by parvinderandroid** [comment]
> 
> 
> In my case I cannot download anything from any site, everything fails. This has been happening to me for a long time, even before moving Firefox Preview to Firefox Nightly.
> 
> I have tried removing all the plugins, checking and authorizing all the permissions, nothing works.
> 
> If I want to download something I have to use Firefox Focus. It's frustrating.
> 
> Xperia Z3 on Android 6.0.1

Oh my god. How the hell did I not notice this? I am the OP of this issue and I am also not able to download anything from any website, not even GitHub.

**Comment by SS1113** [comment]
I have tagged one of the developers working on the download issue let's see what they say.

**Comment by parvinderandroid** [comment]
I am new to bug reporting and don't quite understand much. I am sorry for creating the new issue. @Diana-Rus has already closed that one.

I have updated the title of this issue and edited my comment to reflect the new findings.

**Comment by Diana-Rus** [comment]
Hey, @parvinderandroid  have you checked with for example having been connected only to wi-fi? And also when connected to wi-fi to be more close to the router and then make the download?

**Comment by SS1113** [comment]
@Diana-Rus the user has already confirmed that they can download the files in Chrome browser and other browsers but the files don't download using the latest Nightly and the beta. So I don't think there is an issue with the network connection.

@parvinderandroid I'm trying to reproduce this bug since yesterday but I have found no success. Can you provide any hints on why do you think this could be happening? Perhaps you have changed some settings in you device? Anything that can help us reproduce it. It'll hard to fix this if the developers don't reproduce this on their devices.

**Comment by parvinderandroid** [comment]
Thank you for your response.

I do not want to sound rude, but:

1. I am not connected to Cellular.
2. I am connected to Wi-Fi only.
3. I have a fiber connection that gives a very stable @50 Mbit upload and
download. I am easily able to stream 60 FPS 4K on YouTube. I am attaching a
speedtest here.

**Comment by SS1113** [comment]
@parvinderandroid is your WiFi set to act like a metered connection on you mobile? Check it in your phone's WiFi settings.

**Comment by MinionAttack** [comment]
In my case it does not matter, either by wifi or mobile network, with vpn or without vpn ... With Firefox Focus I can download without any problem whatever the configuration.

**Comment by SS1113** [comment]
@parvinderandroid I can see from the video that you are using a **VPN connection**. Can you try downloading without using the VPN?

**Comment by parvinderandroid** [comment]
> 
> 
> @parvinderandroid I can see from the video that you are using a **VPN connection**. Can you try downloading without using the VPN?

The VPN symbol is due to the presence of 1.1.1.1. I also sometimes turn on Adguard, but that's not the case for this video.

**Comment by Scripterr** [comment]
Fenix did have a issue with ad blockers which maybe interfere with downloads in some devices, your issue seems like this bug https://github.com/mozilla-mobile/fenix/issues/2639 but I'm not sure if it's same case, maybe a developer can confirm if it's same bug as the one I mentioned.

**Comment by Amejia481** [comment]
Unfortunately, I'm unable to reproduce the issue on Pixel 3 with Android 10.
QA please help us to pinpoint which devices may be affected. 

```
Nightly 200706 06:01 (Build #21880609)
AC: 49.0.20200702190156, bafa9299b
GV: 80.0a1-20200702094606
AS: 61.0.7
```

**Comment by sheikh-azharuddin** [comment]
Works fine here in my Asus rog 2.
Can you try from another device? Might be device specific issue for example the new OPPO find x2 has protection feature apart from android default permissions which protects various section like files, clipboard, contacts etc...

**Comment by Amejia481** [comment]
> Nightly 200706 06:01 (Build #21880609)
> AC: 49.0.20200702190156, bafa9299b
> GV: 80.0a1-20200702094606
> AS: 61.0.7

I also tested on Moto g5, but I'm unable to reproduce.
 
```

Nightly 200706 06:01 (Build #21880609)
AC: 49.0.20200702190156, bafa9299b
GV: 80.0a1-20200702094606
AS: 61.0.7
```

**Comment by SS1113** [comment]
I have tested this on three devices;
On **`Nightly 200706 06:01`**

But the issue doesn't reproduce on them as well.

**Motorola One Power Android 10**

**Xioami Mi A2, Android 10**

**Realme C3, Android 10**

**Comment by sheikh-azharuddin** [comment]
@SS1113 I feel this might be related to old android version... The user is also using A9... Can you pls test with some old version android device?

**Comment by SS1113** [comment]
@sheikh-azharuddin Yeah, I do have a device with Android 9 but it's a bit out of my reach but will try to test it there.

**Comment by MinionAttack** [comment]
I have uninstalled the application, I reinstall the application from the Play Store. Without configuring anything I try to download a file, it fails. I try to configure the tracking protection, both by default and strict, in both fails. At all times I am without uBlock Origin, so in this case the ad blocker does not interfere.

Xperia Z3, Android 6.0.1

**Comment by sheikh-azharuddin** [comment]
@MinionAttack you are also running old android version... Saipan will test and let us know.. Lets wait for the result

**Comment by SS1113** [comment]
@sheikh-azharuddin I tested this on **Realme C2, Android 9**, on **`Nightly 200706 06:01`**. But wasn't able to reproduce.

**Comment by sheikh-azharuddin** [comment]
Weird then why some people are facing issues🙄

**Comment by SS1113** [comment]
> Weird then why some people are facing issues🙄

Don't know. This is the reason why I don't buy devices which are heavily skinned/modified. I try to stay as close to Stock Android as possible. The only way to avoid most incompatibilities, I think.

But the developers seem to be adding a feature to get download logs:  https://github.com/mozilla-mobile/android-components/issues/7626 let's see if that'll help.

**Comment by owlishDeveloper** [comment]
Unable to reproduce with Firefox Preview 5.2.0, device Samsung Galaxy S8 (Android 9) and Firefox Nightly 6/30 (same device) - both private and normal tabs

**Comment by sv-ohorvath** [comment]
@Amejia481 We've tried to reproduce the issue on the latest Nightly build 7/7, on the following devices:
Huawei P20 Lite (Android 9), Huawei P9 Lite (Android 7), Samsung Galaxy Note 10(Android 10), Xiaomi Redmi Note 8T (Android 9), Samsung Galaxy Tab S6 (Android 9), HTC Desire 820 (Android 6.0.1)

Unfortunately, we cannot reproduce the issue and have no STR to help with.

**Comment by Amejia481** [comment]
@sv-ohorvath No worries thanks for the help :)

@parvinderandroid @MinionAttack  we are going to introduce some log statement on https://github.com/mozilla-mobile/android-components/pull/7632 to try to determine what is happening.

**Comment by MinionAttack** [comment]
> 
> 
> @sv-ohorvath No worries thanks for the help :)
> 
> @parvinderandroid @MinionAttack we are going to introduce some log statement on [mozilla-mobile/android-components#7632](https://github.com/mozilla-mobile/android-components/pull/7632) to try to determine what is happening.

Ok thanks, I am glad to help with whatever you need.

**Comment by parvinderandroid** [comment]
> 
> 
> @sv-ohorvath No worries thanks for the help :)
> 
> @parvinderandroid @MinionAttack we are going to introduce some log statement on [mozilla-mobile/android-components#7632](https://github.com/mozilla-mobile/android-components/pull/7632) to try to determine what is happening.

Thank you. I feel guilty for posting this and giving so many people such a headache. I don't even download much. I know I  have been late to respond most of the times, but I'll also be glad to help in any way possible.

**Comment by SS1113** [comment]
@parvinderandroid please don't feel any guilt, if anything, you helped so many other users who have not reported this issue and are facing this problem everyday. Thanks you so much for taking your time and helping so many users who might not know where and how to report bugs they face. 

Thank you so much... :)

**Comment by liuche** [comment]
I'm going to close this because QA wasn't able to repro with the Nightly version, and there are string of other comments unable to repro. Please reopen/comment if you can repro.

**Comment by parvinderandroid** [comment]
> 
> 
> I'm going to close this because QA wasn't able to repro with the Nightly version, and there are string of other comments unable to repro. Please reopen/comment if you can repro.

Sir, I am a long time Firefox user and I am still facing the same issue. I am using ADM as and when needed, but it's getting bad now, I can't even save a picture. It always says download failed. I am hopeful the download log entries will help. But how will I know what to do if the case is closed?

**Comment by Amejia481** [comment]
> > I'm going to close this because QA wasn't able to repro with the Nightly version, and there are string of other comments unable to repro. Please reopen/comment if you can repro.
> 
> Sir, I am a long time Firefox user and I am still facing the same issue. I am using ADM as and when needed, but it's getting bad now, I can't even save a picture. It always says download failed. I am hopeful the download log entries will help. But how will I know what to do if the case is closed?

In nightly we added some logs statements that can help to figure out, what is happening @parvinderandroid would you mind [sharing the logcat logs when the download fails](https://developer.android.com/studio/command-line/logcat)?

**Comment by MinionAttack** [comment]
> 
> 
> > > I'm going to close this because QA wasn't able to repro with the Nightly version, and there are string of other comments unable to repro. Please reopen/comment if you can repro.
> > 
> > 
> > Sir, I am a long time Firefox user and I am still facing the same issue. I am using ADM as and when needed, but it's getting bad now, I can't even save a picture. It always says download failed. I am hopeful the download log entries will help. But how will I know what to do if the case is closed?
> 
> In nightly we added some logs statements that can help to figure out, what is happening @parvinderandroid would you mind [sharing the logcat logs when the download fails](https://developer.android.com/studio/command-line/logcat)?

I've never done that, what logging level do you want and how to specify to get only Firefox logs?

**Comment by Amejia481** [comment]
> > > > I'm going to close this because QA wasn't able to repro with the Nightly version, and there are string of other comments unable to repro. Please reopen/comment if you can repro.
> > > 
> > > 
> > > Sir, I am a long time Firefox user and I am still facing the same issue. I am using ADM as and when needed, but it's getting bad now, I can't even save a picture. It always says download failed. I am hopeful the download log entries will help. But how will I know what to do if the case is closed?
> > 
> > 
> > In nightly we added some logs statements that can help to figure out, what is happening @parvinderandroid would you mind [sharing the logcat logs when the download fails](https://developer.android.com/studio/command-line/logcat)?
> 
> I've never done that, what logging level do you want and how to specify to get only Firefox logs?

No worries, to access the logs you need to use a command line tool called [logcat](https://developer.android.com/studio/command-line/logcat) that comes with [adb](https://developer.android.com/studio/command-line/adb).

1. Install adb ([instructions](https://www.xda-developers.com/install-adb-windows-macos-linux/)).
2. Before starting the download run `adb logcat -d > logcat.txt` this will capture all the logs and put them on a file named `logcat.txt`.
3. Try to download again, then the file `logcat.txt` should be populated with the logs.
4. Upload the file `logcat.txt` to this ticket.

If you need any additional help just let me know!

**Comment by MinionAttack** [comment]
Here is my log. I've tried to download the same file several times. I think the lines of the error are 9229 and 11641

[logcat.txt](https://github.com/mozilla-mobile/fenix/files/4995497/logcat.txt)

Regards.

**Comment by parvinderandroid** [comment]
I tried downloading https://github.com/TeamNewPipe/NewPipe/releases


[logcat - before download.txt](https://github.com/mozilla-mobile/fenix/files/4995664/logcat.-.before.download.txt)


[logcat - after download.txt](https://github.com/mozilla-mobile/fenix/files/4995665/logcat.-.after.download.txt)

**Comment by st3fan** [comment]
@parvinderandroid are you running some kind of intercepting VPN on your device?

**Comment by parvinderandroid** [comment]
> 
> 
> @parvinderandroid are you running some kind of intercepting VPN on your device?

I had AdGuard and 1.1.1.1 app installed. AdGuard was never enabled when I used Firefox. I used 1.1.1.1 though.

Before doing this logcat thing, I disabled AdGuard, uninstalled it, and uninstalled its certificate. I didn't enable 1.1.1.1 either. So the logcat files should be pure vanilla.

**Comment by Amejia481** [comment]
@parvinderandroid this is the stack trace of the issue, I will take a look:

```
07-29 20:39:17.333  8021  8079 D AbstractFetchDownloadService: Fetching download
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: Unable to complete download marked as FAILED
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: java.io.FileNotFoundException: /storage/emulated/0/Download/NewPipe_v0.19.8.apk (No such file or directory)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at java.io.FileOutputStream.open0(Native Method)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at java.io.FileOutputStream.open(FileOutputStream.java:308)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at java.io.FileOutputStream.<init>(FileOutputStream.java:238)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at mozilla.components.feature.downloads.AbstractFetchDownloadService.useFileStream$feature_downloads_release(AbstractFetchDownloadService.kt:28)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at mozilla.components.feature.downloads.AbstractFetchDownloadService$performDownload$1.invoke(AbstractFetchDownloadService.kt:15)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at mozilla.components.concept.fetch.Response$Body.useStream(Response.kt:1)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at mozilla.components.feature.downloads.AbstractFetchDownloadService.performDownload$feature_downloads_release(AbstractFetchDownloadService.kt:12)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at mozilla.components.feature.downloads.AbstractFetchDownloadService.startDownloadJob$feature_downloads_release(AbstractFetchDownloadService.kt:2)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at mozilla.components.feature.downloads.AbstractFetchDownloadService$onStartCommand$1.invokeSuspend(AbstractFetchDownloadService.kt:3)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:3)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:21)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:1)
07-29 20:39:17.350  8021  8079 E AbstractFetchDownloadService: 	at 
kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:11)

```

**Comment by ekager** [comment]
It does sound like there's a download issue here for @parvinderandroid.

Also for everyone else having problems installing APKs from Fenix please make sure you've given Fenix permission in your device settings to "install unknown apps" or they won't be able to be opened. 

For me it's Settings -> Apps -> Firefox Preview -> Advanced -> Install Unknown Apps

**Comment by Amejia481** [comment]
@ekager the stack trace points to ac we will take a look I opened  https://github.com/mozilla-mobile/android-components/issues/7921 for it.

**Comment by Amejia481** [comment]
> Here is my log. I've tried to download the same file several times. I think the lines of the error are 9229 and 11641
> 
> [logcat.txt](https://github.com/mozilla-mobile/fenix/files/4995497/logcat.txt)
> 
> Regards.

@MinionAttack I was able to replicated the issue https://www.tdtchannels.com/android/apks/tdtchannels_v2.2.0.apk
 this case is different it looks like the site is writing some cookies and for some reason we are not passing the cookies when downloading.

This is the request on desktop:

<img width="1791" alt="image" src="https://user-images.githubusercontent.com/773158/88872399-0da8b700-d1e8-11ea-8924-f0a5fff61b3e.png">

Fenix request 

<img width="1325" alt="image" src="https://user-images.githubusercontent.com/773158/88872486-447ecd00-d1e8-11ea-9927-04cd1ea28025.png">


This looks related to this geckoview bug https://bugzilla.mozilla.org/show_bug.cgi?id=1530022 but we will do some further investigation to confirm.

@MinionAttack are you able to download [this](https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf) sample pdf ?

**Comment by MinionAttack** [comment]
> 
> 
> > Here is my log. I've tried to download the same file several times. I think the lines of the error are 9229 and 11641
> > [logcat.txt](https://github.com/mozilla-mobile/fenix/files/4995497/logcat.txt)
> > Regards.
> 
> @MinionAttack I was able to replicated the issue https://www.tdtchannels.com/android/apks/tdtchannels_v2.2.0.apk
> this case is different it looks like the site is writing some cookies and for some reason we are not passing the cookies when downloading.
> 
> This is the request on desktop:
> 
> <img alt="image" width="1791" src="https://user-images.githubusercontent.com/773158/88872399-0da8b700-d1e8-11ea-8924-f0a5fff61b3e.png">
> 
> Fenix request
> 
> <img alt="image" width="1325" src="https://user-images.githubusercontent.com/773158/88872486-447ecd00-d1e8-11ea-9927-04cd1ea28025.png">
> 
> This looks related to this geckoview bug https://bugzilla.mozilla.org/show_bug.cgi?id=1530022 but we will do some further investigation to confirm.
> 
> @MinionAttack are you able to download [this](https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf) sample pdf ?

:O Yes, I'm able to download that dummy pdf file and open it after the download.

**Comment by Amejia481** [comment]
@MinionAttack  could you share the logs? :)

**Comment by MinionAttack** [comment]
@Amejia481 Hi, I've downloaded again the file before sendind the logcat.

[logcat.txt](https://github.com/mozilla-mobile/fenix/files/5000541/logcat.txt)

**Comment by Amejia481** [comment]
It looks like the app is having trouble opening the file this could be the issue: 
`07-30 12:09:48.458 26605 26605 D PdfViewer: File length in MB: -1`

We are going to track this one here https://github.com/mozilla-mobile/fenix/issues/12451

```
07-30 12:09:48.385 26605 26605 D PdfViewerActivity: Activity create/fresh File dummy.pdf (application/pdf) @content://org.mozilla.fenix.feature.downloads.fileprovider/Download/Download/dummy.pdf/null

07-30 12:09:48.400 26605 26605 V ContentUriOpener: Use content type null (declared was application/pdf)

07-30 12:09:48.402 26605 26605 V ViewerManager: LC Look for fragment Viewer #0 null

07-30 12:09:48.413 26605 26605 V PdfViewer: Lifecycle: Saved arg Bundle[{n=dummy.pdf, t=PDF, po=ContentOpenable [application/pdf]: content://org.mozilla.fenix.feature.downloads.fileprovider/Download/Download/dummy.pdf / @null, uri=content://org.mozilla.fenix.feature.downloads.fileprovider/Download/Download/dummy.pdf}]

07-30 12:09:48.413 26605 26605 V PdfViewer: Lifecycle: onCreate

07-30 12:09:48.413 26605 26605 V ViewerManager: Create and feed a new Viewer 0/2131361921

07-30 12:09:48.416 26605 26605 V ImpressionsTracker: [PDF_VIEWER] trackEvent null

07-30 12:09:48.445 26605 26605 V ContentUriOpener: Use content type null (declared was application/pdf)

07-30 12:09:48.445 26605 26605 V PdfViewer: Lifecycle: onCreateView null

07-30 12:09:48.451 26605 26605 V Screen  : 10dp = 30; dpi = 480

07-30 12:09:48.456 26605 26605 V PdfViewer: Lifecycle: onActivityCreated NO_VIEW

07-30 12:09:48.457 26605 26629 I art     : Rejecting re-init on previously-failed class java.lang.Class<com.google.android.gms.org.conscrypt.Java7ExtendedSSLSession>

07-30 12:09:48.457 26605 26629 I art     : Rejecting re-init on previously-failed class java.lang.Class<com.google.android.gms.org.conscrypt.Java7ExtendedSSLSession>

07-30 12:09:48.457 26605 26629 I art     : Rejecting re-init on previously-failed class java.lang.Class<com.google.android.gms.org.conscrypt.Java8ExtendedSSLSession>

07-30 12:09:48.458 26605 26605 I AppLifecycleTracker: App transition to foreground

07-30 12:09:48.458 26605 26629 I art     : Rejecting re-init on previously-failed class java.lang.Class<com.google.android.gms.org.conscrypt.Java8ExtendedSSLSession>

07-30 12:09:48.458 26605 26605 D PdfViewer: File length in MB: -1

07-30 12:09:48.460 26605 26605 D PdfConnection: Connecting to service content://org.mozilla.fenix.feature.downloads.fileprovider/Download/Download/dummy.pdf

07-30 12:09:48.472   923  3857 I ActivityManager: Start proc 26633:com.google.android.apps.pdfviewer/u0i1 for service com.google.android.apps.pdfviewer/com.google.android.apps.viewer.pdflib.PdfDocumentService

07-30 12:09:48.486 26605 26605 V PdfViewer: Lifecycle: Got contents (delayed)PDF

07-30 12:09:48.486 26605 26605 V PdfViewer: Lifecycle: onStart 

07-30 12:09:48.486 26605 26605 V PdfViewer: Lifecycle: onEnter

07-30 12:09:48.486 26605 26605 D PdfConnection: Connecting to service content://org.mozilla.fenix.feature.downloads.fileprovider/Download/Download/dummy.pdf
```

**Comment by Amejia481** [comment]
Thanks all of you for providing us all the info!

**Comment by MinionAttack** [comment]
@Amejia481 

"It looks like the app is having trouble opening the file this could be the issue"

Maybe I don't get it what you mean, but I can download and open the pdf file. But for the download of the apk of the yesterday's log fails in the download.

**Comment by Amejia481** [comment]
> @Amejia481
> 
> "It looks like the app is having trouble opening the file this could be the issue"
> 
> Maybe I don't get it what you mean, but I can download and open the pdf file. But for the download of the apk of the yesterday's log fails in the download.

What I meant was that this log statement could be the key for finding out what is causing the issue  `07-30 12:09:48.458 26605 26605 D PdfViewer: File length in MB: -1`, this came from the pdf reader app, it saying that it is not getting the file length,  we need to investigate more to fix the issue we are going to track it here #12451 as it's a different issue from this one.

**Comment by MinionAttack** [comment]
@Amejia481 ah ok ok sorry for my confusion Thanks for the clarification.

**Comment by Amejia481** [comment]
No worries, It's not a problem happy to help!

**Comment by parvinderandroid** [comment]
I solved my problem.


After 2 factory resets and way too much testing, I finally ducking figured it out.


If the 'Download' folder does not exist in root of internal memory, download fails.


If I manually create a folder named 'Download' in internal memory, download works.


If I then delete 'Download' folder, download fails again.


Video of me demonstrating this: https://i.imgur.com/VJ3cyCr.mp4

**Comment by Amejia481** [comment]
That's great news 🎉 . Anyway we have to investigate why in your device when we try to write a file, the 'Download' folder didn't auto generate. Would you mind sharing the brand, model and OS version that you are using?

**Comment by Amejia481** [comment]
Thanks for creating https://github.com/mozilla-mobile/fenix/issues/13134

**Comment by parvinderandroid** [comment]
> That's great news 🎉 . Anyway we have to investigate why in your device when we try to write a file, the 'Download' folder didn't auto generate. Would you mind sharing the brand, model and OS version that you are using?

### DEVICE INFORMATION

* Android device: Huawei Honor 8 Pro (DUK-L09) - Android 9.0
* Fenix version: 79.1.0-beta.1 & Nightly 200730 06:21

Bromite, Chrome and ADM are able to create the folder if it doesn't exist.

**Comment by parvinderandroid** [comment]
> Thanks for creating #13134

Always ready to help.

On a side note, I am very happy I figured it out, can I add this to my resume? /s

**Comment by Amejia481** [comment]
> > Thanks for creating #13134
> 
> Always ready to help.
> 
> On a side note, I am very happy I figured it out, can I add this to my resume? /s

I think you can :)

**Comment by FrostedIce339** [comment]
> 
> 
> Thanks for creating #13134

@parvinderandroid @Amejia481 I am able to reproduce the issue. I tried deleting the Download folder in my device's internal storage and tried downloading an APK from Apkmirror.com and the download failed with 'Download failed' with 'Try again' dialogue and if iI press dialogue it fails again. And If the Download folder is available, I am able to download files.

**Comment by Amejia481** [comment]
Thanks @FrostedIce339 for reporting. The info is pretty valuable.

> I am also getting a 'null download failed' dialogue and the notification cannot be dismissed and I have to Force stop the Firefox app to make it disappear from the notification bar.

Would you mind opening a ticket describing the steps to simulate the issue? Thanks in advance!

> The Download folder gets created if I download a link (by long pressing on it and clicking 'Download link') but the Download folder doesn't get created if I try to download an APK file from the Apkmirror.com.

Awesome discovering 🕵️!

**Comment by trefmanic** [comment]
In my case, I get a notification "Download failed", but files are actually appearing inside my default download folder.

Nightly 200823 06:02 (Build #2015759603)
AC: 56.0.20200821184145, 96a8a14dc
GV: 81.0a1-20200820093209
AS: 61.0.12
Running on Android 10.

**Comment by cadeyrn** [comment]
The issue you see is probably #14052, a current regression that already has fix in progress.

**Comment by liuche** [comment]
Will dupe this to #14052, and please check Nightly tomorrow for this change.
